src/landing_page/src/pages/landing/landing.js

```diff
@@ -49,54 +48,66 @@ function Landing() {
   useEffect(() => {
     // Fetch the DMP ID metadata from the DMPHub
     let api = new DmpApi();
-
-    fetch(api.getUrl(), api.getOptions())
-      .then((resp) => {
-        api.handleResponse(resp);
-        return resp.json();
-      })
-      .then((data) => {
-        // console.log(data.items[0].dmp);
+    const controller = new AbortController();
+    const fetchData = async () => {
+      try{
+        const response = await fetch(api.getUrl(),api.getOptions(),{signal:controller.signal});
+        api.handleResponse(response);
+        const data = await response.json();
```

Comment:
So the AbortController will handle things like timeouts?

I need to read up on how await and promises in general work

Reply:
The useEffect cleanup function is generally used to tidy up our code before the component unmounts. The cleanup function can prevent memory leaks.